events-api.go

```diff
@@ -120,26 +129,48 @@ func (e *EventsApi) handleShutdown(stop context.CancelFunc) {
 }
 
 func NewEventsApi(apiClient *openapi.APIClient, options *KableOptions) *EventsApi {
-	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
-
 	eventsApi := &EventsApi{
 		api:     apiClient,
 		options: options,
-		ctx:     ctx,
-		queue:   []Event{},
 	}
 
-	go eventsApi.scheduleFlushQueue()
-	go eventsApi.handleShutdown(stop)
+	// if the queue size is less than 2, then recording any number of events will result
+	// in calling flush, so there will be no need to initiate the queue, as it won't be used.
+	// also, as flush won't be needed, then there's no need
+	if options.MaxQueueSize > 1 {
+		// initiate the queue with the max queue size specified, so that reporting any
+		// number of events below that number will not result in allocating a new queue.
+		eventsApi.queue = make([]Event, 0, options.MaxQueueSize)
+	}
+
+	// only start the scheduled goroutines if the queue was initialized
+	if eventsApi.queue != nil {
+		ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
+		eventsApi.ctx = ctx
 
-	// Listening to the OS Signals
+		go eventsApi.scheduleFlushQueue()
+		go eventsApi.handleShutdown(stop)
+	}
 
 	return eventsApi
 }
 
 func (e *EventsApi) EnqueueEvent(events ...Event) {
+	// if queue size is less than 2, then no need to add the events to the queue, so send
+	// them directly and return(no need to clear the queue, as it must be already empty).
+	if e.options.MaxQueueSize < 2 {
+		e.sendEvents(events)
```

Comment:
@asmsh Our linting action is failing here since the return value is not checked. Could you make that change? Other than that, this PR looks great.